openmc/tallies.py

```diff
@@ -1482,19 +1482,30 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
```

Comment:
I actually like the use of a copy here, since in general, I think it is best to leave intact the original `Tally` objects to which the user has access. I assume that these `Tally` copies will be deleted in memory after the arithmetic operation is complete? Although this change will temporarily require more memory to store the `Tally` copies than was previously the case, the memory footprint following the tally arithmetic operation will be the same, so I think this behavior is fine.